OnlineStore/store/src/main/java/RandomStorePopulator.java

```diff
@@ -1,34 +1,70 @@
-                Class[] params = {String.class};
-                myCategory = (Category) categoryClass.getConstructor(params).newInstance(faker.commerce().department());
-            } catch (ClassNotFoundException | IllegalAccessException | InstantiationException | NoSuchMethodException | InvocationTargetException e) {
-                e.printStackTrace();
-            }
-            Product product = new Product.Builder()
-                    .withName(faker.commerce().productName())
-                    .withCategory(myCategory)
-                    //.withCategory(new Category(faker.commerce().department()))
-                    .withPrice(Double.valueOf(faker.commerce().price()))
-                    .withQuantity(faker.number().numberBetween(0,30))
-                    .withRating(faker.number().numberBetween(0, 11))
-                    .build();
-            products.add(product);
+
+    public List<String> getNameOfCategoryClasses(){
+        for (int j = 0; j < le; j++){
+            nameOfCategoryClasses.add(subClassesForCategory.parallelStream().collect(Collectors.toList())
+                    .get(j).getName());
         }
+        return nameOfCategoryClasses;
+    }
+
+    public List<Product> populateStore(int quantity){
+        List<Product> products = new ArrayList<>();
+        /*if (quantity > store.getCapacity()){
+            System.out.println("The store capacity is less than quantity of products, which you try to generate");
+        } else {*/
+            String productName = "";
+            String className = "";
+            for (int i = 0; i < quantity; i++) {
+                int randomInt = (int) (Math.random() * le);
+                className = getNameOfCategoryClasses().get(randomInt);
+                productName = getProductNameByCategoryName(className);
+                Product product = null;
+                try {
+                    product = new Product.Builder()
+                            .withName(productName)
+                            .withCategory((Category) Class.forName(getNameOfCategoryClasses().get(randomInt))
+                                    .newInstance())
+                            .withPrice(Double.valueOf(faker.commerce().price()))
+                            .withQuantity(faker.number().numberBetween(0, 30))
+                            .withRating(faker.number().numberBetween(0, 11))
+                            .build();
+                } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+                    e.printStackTrace();
+                }
+                products.add(product);
+            }
+        /*}*/
         return products;
     }
 
+    public String getProductNameByCategoryName(String categoryNameField){
+        String name = categoryNameField.trim().toLowerCase();
+        String productName = "";
+        if (name.equals("categories.fruit")){
```

Comment:
1/ you can use switch here, 2. please use class oblects as is to get there full names, like Fruit.class.getCanonicalName or so